src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -569,9 +569,9 @@ public function load(array $configs, ContainerBuilder $container): void
             $container->removeDefinition('console.command.scheduler_debug');
         }
 
-        // messenger depends on validation being registered
+        // messenger depends on validation, and lock being registered
         if ($messengerEnabled) {
-            $this->registerMessengerConfiguration($config['messenger'], $container, $loader, $this->readConfigEnabled('validation', $container, $config['validation']), $this->readConfigEnabled('lock', $container, $config['lock']));
+            $this->registerMessengerConfiguration($config['messenger'], $container, $loader, $this->readConfigEnabled('validation', $container, $config['validation']), $this->readConfigEnabled('lock', $container, $config['lock']) && ($config['lock']['resources']['default'] ?? false));
```

Comment:
I don't think this is the correct solution, you should be able to define a different lock factory, e.g.:

```yaml
framework:
    lock:
        deduplicate: '%env(LOCK_DEDUPLICATE_DSN)%'

    messenger:
        buses:
            messenger.bus.default:
                middleware:
                    - deduplicate_middleware: [ lock.deduplicate.factory ]
```

However, this doesn't seem to work either:

> Invalid middleware factory "deduplicate_middleware": a middleware factory must be an abstract definition.  

In any case, this seems like a different bug that needs fixing. I can look into it later today. Once that's fixed, this solution won't make much sense anymore, since the lock name won't need to be `default`.

Reply:
Ok, I went over the feature more closely. The middleware is enabled by default, so you don't need to add it manually. It would still make sense to me if the lock factory was configurable, but that would be a new feature.